src/app/InteractionModelEngine.h

```diff
@@ -373,6 +373,22 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
                                  System::PacketBufferHandle && aPayload) override;
     void OnResponseTimeout(Messaging::ExchangeContext * ec) override;
 
+    /**
+     * This parses the attribute path list to ensure it is well formed. If so, for each path in the list, it will expand to a list
+     * of concrete paths and walk each path to check if it has privileges to read that attribute.
+     *
+     * If there is AT LEAST one concrete path that has sufficient privilege, aHasValidAttributePath will be set to true. Otherwise,
+     * it will be set to false.
+     *
+     * aRequestedAttributePathCount will be updated to reflect the number of attribute paths in the request.
+     *
+     * @retval CHIP_NO_ERROR if no error is encountered.
+     *
+     */
+    CHIP_ERROR ParseAttributePaths(Access::SubjectDescriptor & aSubjectDescriptor,
```

Comment:
Does this use any member state?  If not, can we make this static?

Reply:
yes, should be static.

Reply:
Done, thanks